src/agent/Core/SpawningKit/UserSwitchingRules.h

```diff
@@ -56,6 +56,9 @@ struct UserSwitchingInfo {
 	gid_t gid;
 	int ngroups;
 	shared_array<gid_t> gidset;
+
+	struct passwd lve_userpwd, *lve_userpwd_complete;
+	shared_array<char> lve_userpwd_strbuf;
```

Comment:
To avoid ambiguity with C++11's standard library, please write `boost::shared_array`, `boost::shared_ptr` and `boost::make_shared`. Otherwise compilation will fail on e.g. OS X.